app.js

```diff
@@ -2,6 +2,9 @@ angular.module("App",['main']);
 
 angular.module("main",[]);
 
- angular.module("main").controller("mainController",function($scope){
-    // Controller body
+angular.module("main").controller("mainController",function($scope, calculator){
+    $scope.pattern = new RegExp("^£?[0-9]+.?[0-9]*p?$");
```

Comment:
does £2x is a valid input?
does 1.0p is a valid input?
none of this should be valid